src/r_tipping/RateSystem.jl

```diff
@@ -0,0 +1,133 @@
+# Consider the ODE dxₜ/dt = f(xₜ,p). We want to ramp one parameter of this ODE.
+"""
+    RateConfig
+
+Time-dependent forcing protocol ``p(t)`` describing the evolution of a parameter over a
+time interval `[section_start, section_end]`.
+Used to construct a non-autonomous `RateSystem`.
+
+## Fields
+- `pfunc`: function ``p(t)`` from ``ℝ → ℝ`` describing the parameter time dependence
+- `section_start`: start of the time interval over which `pfunc` is considered
+- `section_end`: end of the time interval over which `pfunc` is considered
+
+## Description
+The `RateConfig` type allows to specify the functional form of a parametric
+forcing over a time interval. This forcing protocol can then be applied to the parameter
+of a dynamical system using the `RateSystem` constructor, which also allows to modify
+the rate and magnitude of the forcing.
+"""
+mutable struct RateConfig
+    pfunc::Function
+    section_start::Float64
+    section_end::Float64
+end
+
```

Comment:
After this PR is merged, I will contribute a `reverse = false` option to the config. If true, the forcing is reversed in the timewindow `Δt, 2Δt` using exactly the same functional form used to scale it.